dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -53,9 +52,11 @@
     protected boolean isRunning = false;
 
     /** The transfer function for regular nodes. */
+    // TODO: make final. Currently, the transferFunction has a reference to the analysis, so it
```

Comment:
I add more info to this TODO, which I think is the problem.  But please have a look.

Reply:
Thanks a lot for cherry-picking the unrelated changes! This expanded explanation is good.